test/udp_tracker.cpp

```diff
@@ -171,6 +173,7 @@ struct udp_tracker
 	{
 		m_socket.cancel();
 		m_socket.close();
+		m_ios.stop();
```

Comment:
My theory is that since `m_socket.close()` cancel the pending task in the `m_ios`, this `run()` return once the last queued `on_udp_receive` finish, but since the last line is another async task schedule, somehow (and this is pure speculation), somehow there is a synchronization problem and `this` become invalid at the moment of `bind`.

Well, I can not prove it, reproduce it or even properly navigate the boost code, so I rather not do anything about it at this moment.